api/src/main/java/io/opencensus/metrics/DoubleGauge.java

```diff
@@ -0,0 +1,236 @@
+ * limitations under the License.
+ */
+
+package io.opencensus.metrics;
+
+import io.opencensus.internal.Utils;
+import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Double Gauge metric, to report instantaneous measurement of a double value. Gauges can go both up
+ * and down. The gauges values can be negative. For example, Free memory or Total memory. See {@link
+ * io.opencensus.metrics.MetricRegistry} for an example of its use.
+ *
+ * <p>Example 1: Create a Gauge with default labels.
+ *
+ * <pre>{@code
+ * class YourClass {
+ *
+ *   private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+ *   DoubleGauge totalMemory = metricRegistry.addDoubleGauge(
+ *       "Total_memory", "Total CPU Memory", "1", new ArrayList<LabelKey>());
+ *
+ *   Point defaultPoint = totalMemory.getDefaultTimeSeries();
+ *
+ *   void doWork() {
+ *      defaultPoint.inc();
+ *      // Your code here.
+ *      defaultPoint.dec();
+ *   }
+ * }
+ *
+ * }</pre>
+ *
+ * <p>Example 2: You can also use labels(keys and values) to track different types of metric.
+ *
+ * <pre>{@code
+ * class YourClass {
+ *
+ *   private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+ *
+ *   List<LabelKey> keys = Collections.singletonList(LabelKey.create("Type","desc"));
+ *   DoubleGauge totalMemory = metricRegistry.addDoubleGauge(
+ *       "Total_Memory", "Total CPU Memory", "1", keys);
+ *
+ *   List<LabelValue> usedMemory = Collections.singletonList(LabelValue.create("Used"));
+ *   Point point = totalMemory.getOrCreateTimeSeries(usedMemory);
+ *
+ *   void doSomeWork() {
+ *      point.set(12.5);
+ *      // Your code here.
+ *      point.dec();
+ *   }
+ * }
+ * }</pre>
+ *
+ * @since 0.17
+ */
+@ThreadSafe
+public abstract class DoubleGauge {
+
+  /**
+   * Creates a TimeSeries and returns a {@code Point}, which is part of the TimeSeries. This is more
+   * convenient form when you can want to manually increase and decrease values as per your service
+   * requirements. The number of label values must be the same to that of the label keys passed to
+   * {@link MetricRegistry#addDoubleGauge}.
+   *
+   * @param labelValues the list of label values.
+   * @return a {@code Point} the value of single gauge.
+   * @since 0.17
+   */
+  public abstract Point getOrCreateTimeSeries(List<LabelValue> labelValues);
+
+  /**
+   * Returns a {@code Point} for a gauge with all labels not set, or default labels.
+   *
+   * @return a {@code Point} the value of default gauge.
+   * @since 0.17
+   */
+  public abstract Point getDefaultTimeSeries();
+
+  /**
+   * Removes the {@code TimeSeries} from gauge metric, if it is present.
+   *
+   * @param labelValues the list of label values.
+   * @since 0.17
+   */
+  public abstract void removeTimeSeries(List<LabelValue> labelValues);
+
+  /**
+   * Removes all {@code TimeSeries}s from gauge metric.
```

Comment:
We should say that references to previous Point object are invalid (not part of the metric).